lib/rack/cache/redis_entitystore.rb

```diff
@@ -1,34 +1,93 @@
 require 'rack/cache/entity_store'
 require 'redis-rack-cache/constants'
 require 'rack/cache/redis_base'
+require 'zlib'
+require 'zlib/gzip_compression'
 
 module Rack
   module Cache
     class EntityStore
       class Redis < self
+        MINIMUM_COMPRESSION_BYTESIZE = 1_000
+
         include RedisBase
 
         def exist?(key)
           cache.exists key
         end
 
         def read(key)
-          cache.get key
+          decompress cache.get(key)
         end
 
         def write(body, ttl=0)
           buf = StringIO.new
-          key, size = slurp(body) {|part| buf.write(part) }
+          key, _size = slurp(body) { |part| buf.write(part) }
           ttl = ttl.to_i.zero? ? default_ttl : ttl
+          value = compress decompress(buf.string)
```

Comment:
why did i have to do this? who knows! but it was the key to getting shit to work properly...